the package deriving is broken with OCaml 4.03.0

# opam switch
    4.03.0        C 4.03.0        Official 4.03.0 release
    # opam --version
    1.2.2
    # nohup opam install -v deriving

nohup.out can be found here:
https://gist.github.com/UnixJunkie/e86571930e6757314cc0dad429e91d93

The shorter log is


   # opam install deriving
   The following actions will be performed:
     - install deriving 0.7
   
   =-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
   [deriving] Archive in cache
   
   =-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
   [ERROR] The compilation of deriving failed at "make".
   Processing  1/1: [deriving: ocamlfind remove]
   #=== ERROR while installing deriving.0.7 ======================================#
   # opam-version 1.2.2
   # os           linux
   # command      make
   # path         /home/berenger/.opam/4.03.0/build/deriving.0.7
   # compiler     4.03.0
   # exit-code    2
   # env-file     /home/berenger/.opam/4.03.0/build/deriving.0.7/deriving-30913-d2c37b.env
   # stdout-file  /home/berenger/.opam/4.03.0/build/deriving.0.7/deriving-30913-d2c37b.out
   # stderr-file  /home/berenger/.opam/4.03.0/build/deriving.0.7/deriving-30913-d2c37b.err
   ### stdout ###
   # Command line: camlp4 '-I' '/home/berenger/.opam/4.03.0/lib/ocaml/camlp4' '-I' '/home/berenger/.opam/4.03.0/lib/optcomp' '-I' '/home/berenger/.opam/4.03.0/lib/ocaml/camlp4' '-I' '/home/berenger/.opam/4.03.0/lib/ocaml/camlp4' '-parser' 'o' '-parser' 'op' '-printer' 'p' 'optcomp.cma' '-parser' 'Camlp4QuotationCommon' '-parser' 'Camlp4OCamlOriginalQuotationExpander' '-parser' 'Camlp4GrammarParser' 'syntax/common/pa_deriving_common.cma' 'syntax/std/pa_deriving_std.cma' 'syntax/classes/show_class.ml' > /tmp/ocamlppf11383
   # [...]
   # Command exited with code 2.
   # + ocamlfind ocamlopt unix.cmxa -I /home/berenger/.opam/4.03.0/lib/ocamlbuild /home/berenger/.opam/4.03.0/lib/ocamlbuild/ocamlbuildlib.cmxa -linkpkg myocamlbuild.ml /home/berenger/.opam/4.03.0/lib/ocamlbuild/ocamlbuild.cmx -o myocamlbuild
   # File "myocamlbuild.ml", line 507, characters 43-62:
   # Warning 3: deprecated: Ocamlbuild_plugin.String.uncapitalize
   # Use String.uncapitalize_ascii instead.
   # File "myocamlbuild.ml", line 520, characters 51-70:
   # Warning 3: deprecated: Ocamlbuild_plugin.String.uncapitalize
   # Use String.uncapitalize_ascii instead.
   # Makefile:7: recipe for target 'build' failed
   ### stderr ###
   # [...]
   # File "setup.ml", line 5927, characters 31-50:
   # Warning 3: deprecated: String.uncapitalize
   # Use String.uncapitalize_ascii instead.
   # File "setup.ml", line 5928, characters 31-48:
   # Warning 3: deprecated: String.capitalize
   # Use String.capitalize_ascii instead.
   # W: Cannot find source file matching module 'pa_deriving_common' in library pa_deriving_common
   # ocamlfind: Package `wikidoc' not found
   # E: Failure("Command ''/home/berenger/.opam/4.03.0/bin/ocamlbuild' syntax/common/pa_deriving_common.cma syntax/common/pa_deriving_common.cmxa syntax/common/pa_deriving_common.a syntax/common/pa_deriving_common.cmxs syntax/std/pa_deriving_std.cma syntax/std/pa_deriving_std.cmxa syntax/std/pa_deriving_std.a syntax/std/pa_deriving_std.cmxs syntax/classes/pa_deriving_classes.cma syntax/classes/pa_deriving_classes.cmxa syntax/classes/pa_deriving_classes.a syntax/classes/pa_deriving_classes.cmxs lib/deriving.cma lib/deriving.cmxa lib/deriving.a lib/deriving.cmxs lib/deriving_num.cma lib/deriving_num.cmxa lib/deriving_num.a lib/deriving_num.cmxs -use-ocamlfind -tag debug' terminated with error code 10")
   # make: *** [build] Error 1
   
   
   
   =-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
   The following actions failed
     - install deriving 0.7
   No changes have been performed

--- a/packages/deriving/deriving.0.7.1/descr
+++ b/packages/deriving/deriving.0.7.1/descr
@@ -1,0 +1,1 @@
+Extension to OCaml for deriving functions from type declarations

--- a/packages/deriving/deriving.0.7.1/findlib
+++ b/packages/deriving/deriving.0.7.1/findlib
@@ -1,0 +1,1 @@
+deriving

--- a/packages/deriving/deriving.0.7.1/opam
+++ b/packages/deriving/deriving.0.7.1/opam
@@ -1,0 +1,24 @@
+opam-version: "1.2"
+name: "deriving"
+version: "0.7.1"
+maintainer: "dev@ocsigen.org"
+author: "Jeremy Yallop <yallop@gmail.com>"
+homepage: "http://github.com/ocsigen/deriving/"
+bug-reports: "https://github.com/ocsigen/deriving/issues/"
+dev-repo: "https://github.com/ocsigen/deriving.git"
+license: "MIT"
+build: [
+  ["./configure" "--prefix" prefix "--%{type_conv:enable}%-tc"]
+  [make]
+]
+install: [make "install"]
+remove: ["ocamlfind" "remove" "deriving"]
+depends: [
+  "ocamlfind"
+  "base-bytes"
+  "optcomp" {>= "1.6"}
+  "camlp4"
+  "oasis" {build & >= "0.4.4"}
+]
+depopts: ["type_conv"]
+conflicts: ["type_conv" {< "108.07.00"}]

--- a/packages/deriving/deriving.0.7.1/url
+++ b/packages/deriving/deriving.0.7.1/url
@@ -1,0 +1,2 @@
+archive: "https://github.com/ocsigen/deriving/archive/0.7.1.tar.gz"
+checksum: "1260cf1427560542ae5725a6871775f0"

--- a/packages/deriving/deriving.0.7/opam
+++ b/packages/deriving/deriving.0.7/opam
@@ -1,11 +1,18 @@
 opam-version: "1.2"
+name: "deriving"
+version: "0.7"
 maintainer: "dev@ocsigen.org"
+author: "Jeremy Yallop <yallop@gmail.com>"
+homepage: "http://github.com/ocsigen/deriving/"
+bug-reports: "https://github.com/ocsigen/deriving/issues/"
+dev-repo: "https://github.com/ocsigen/deriving.git"
+license: "MIT"
 build: [
   ["ocaml" "setup.ml" "-configure" "--prefix" prefix "--%{type_conv:enable}%-tc"]
   [make]
-  [make "install"]
 ]
-remove: [["ocamlfind" "remove" "deriving"]]
+install: [make "install"]
+remove: ["ocamlfind" "remove" "deriving"]
 depends: [
   "ocamlfind"
   "base-bytes"
@@ -15,3 +22,4 @@ depends: [
 ]
 depopts: ["type_conv"]
 conflicts: ["type_conv" {< "108.07.00"}]
+available: [ocaml-version <= "4.02.3"]

